subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -709,8 +712,72 @@ static u32_t isr_rx_adv_sr_report(struct pdu_adv *pdu_adv_rx, u8_t rssi_ready)
 }
 #endif /* CONFIG_BLUETOOTH_CONTROLLER_SCAN_REQ_NOTIFY */
 
-static inline u32_t isr_rx_adv(u8_t devmatch_ok, u8_t irkmatch_ok,
-			       u8_t irkmatch_id, u8_t rssi_ready)
+static inline bool isr_adv_sr_check(struct pdu_adv *pdu, u8_t devmatch_ok,
+				    u8_t irkmatch_ok, u8_t irkmatch_id)
+{
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	return ((((_radio.advertiser.filter_policy & 0x01) == 0) &&
+		 ctrl_rl_allowed(pdu->tx_addr,
+				 pdu->payload.scan_req.scan_addr)) ||
+		(devmatch_ok) || (ctrl_rl_enabled() && irkmatch_ok &&
+				  ctrl_irk_whitelisted(irkmatch_id))) &&
+		(1 /** @todo own addr match check */);
+#else
+	return (((_radio.advertiser.filter_policy & 0x01) == 0) &&
+		(devmatch_ok)) &&
+		(1 /** @todo own addr match check */);
+#endif
+}
+
+static inline bool isr_adv_tgta_check(struct pdu_adv *adv, struct pdu_adv *ci,
+				      u8_t irkmatch_ok, u8_t irkmatch_id)
+{
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	if (ctrl_rl_enabled() && irkmatch_ok) {
+		return ctrl_rl_idx_match(irkmatch_id, _radio.advertiser.rl_idx);
+#else
+	if (0) {
+#endif
+	} else {
+		return !memcmp(adv->payload.direct_ind.tgt_addr,
+			ci->payload.connect_ind.init_addr,
+			BDADDR_SIZE);
+	}
+}
```

Comment:
The whole #else and else { branches seem unnecessary since you return from within the first if-branch. I.e. the "return !memcmp(..." can be on the top level of the function as far as I can see.